internal/pkg/csumio/csumreader_test.go

```diff
@@ -0,0 +1,191 @@
+package csumio_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cybozu-go/fin/internal/pkg/csumio"
+)
+
+func TestChecksumReader_ChecksumMismatch_VerificationEnabled(t *testing.T) {
+	// Description:
+	// Check that checksum verification detects checksum mismatch when verification is enabled
+	//
+	// Arrange:
+	// - Data file contains test data
+	// - Checksum file contains checksum for different data (intentionally wrong)
+	// - Checksum verification is enabled
+	//
+	// Act:
+	// Read data using ChecksumReader
+	//
+	// Assert:
+	// Should fail with ErrChecksumMismatch
+
+	chunkSize := 4096 // 4KiB
+	cases := []struct {
+		name string
+		data []byte
+	}{
+		{name: "exact_chunk", data: bytes.Repeat([]byte("a"), 4096)},     // exactly one chunk
+		{name: "partial_chunk", data: bytes.Repeat([]byte("a"), 4095)},   // less than one chunk
+		{name: "more_than_chunk", data: bytes.Repeat([]byte("a"), 4097)}, // more than one chunk
+	}
```

Comment:
Please add one more test case for the data size 8192 bytes. It's to verify reading multiple 4096 byte chanks can be read or not.